apps/builder/app/builder/features/style-panel/sections/transitions/transition-property.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState, useEffect } from "react";
+import { animatableProperties } from "@webstudio-is/css-data";
+import {
+  Label,
+  InputField,
+  Combobox,
+  ComboboxAnchor,
+  useCombobox,
+  ComboboxContent,
+  ComboboxLabel,
+  ComboboxListbox,
+  ComboboxListboxItem,
+  ComboboxSeparator,
+  theme,
+  css,
+  NestedInputButton,
+} from "@webstudio-is/design-system";
+import type { KeywordValue, TupleValueItem } from "@webstudio-is/css-engine";
+
+type AnimatableProperties = (typeof animatableProperties)[number];
+type NameAndLabel = { name: string; label?: string };
+type TransitionPropertyProps = {
+  property: KeywordValue;
+  onPropertySelection: (property: TupleValueItem) => void;
+};
+
+const commonPropertiesSet = new Set<AnimatableProperties>([
+  "all",
+  "opacity",
+  "margin",
+  "padding",
+  "border",
+  "transform",
+  "filter",
+  "flex",
+  "background-color",
+]);
+
+const comboBoxStyles = css({ zIndex: theme.zIndices[1] });
+
+export const TransitionProperty = ({
+  property,
+  onPropertySelection,
+}: TransitionPropertyProps) => {
+  const [inputValue, setInputValue] = useState(property?.value ?? "all");
+  useEffect(() => setInputValue(property.value), [property]);
+
+  const {
+    items,
+    isOpen,
+    getComboboxProps,
+    getToggleButtonProps,
+    getInputProps,
+    getMenuProps,
+    getItemProps,
+  } = useCombobox<NameAndLabel>({
+    items: animatableProperties.map((prop) => ({
+      name: prop,
+      label: prop,
+    })),
+    value: { name: inputValue, label: inputValue },
+    selectedItem: undefined,
+    itemToString: (value) => value?.name ?? "",
+    onItemSelect: (prop) => {
+      setInputValue(prop.name);
+      onPropertySelection({ type: "keyword", value: prop.name });
+    },
+    onInputChange: (value) => {
+      if (value === undefined) {
+        return "";
```

Comment:
this return looks weird, I don't think the return value is used, did you mean to just return; ?
```suggestion
        return;
```